templates/chat/src/components/ChatInput.tsx

```diff
@@ -0,0 +1,230 @@
+import { FormEvent, useCallback, useEffect, useLayoutEffect, useRef } from 'react'
+import { DefaultSpinner, TldrawUiTooltip } from 'tldraw'
+import { useChatInputState } from '../hooks/useChatInputState'
+import { ChatInputImage } from './ChatInputImage'
+import { ImageIcon } from './icons/ImageIcon'
+import { SendIcon } from './icons/SendIcon'
+import { UploadIcon } from './icons/UploadIcon'
+import { WhiteboardIcon } from './icons/WhiteboardIcon'
+import { WhiteboardImage, WhiteboardModal } from './WhiteboardModal'
+
+interface ChatInputProps {
+	onSendMessage: (message: string, images: WhiteboardImage[]) => void
+	waitingForResponse: boolean
+	scrollToBottom: (behavior?: ScrollBehavior) => void
+	state: ReturnType<typeof useChatInputState>[0]
+	dispatch: ReturnType<typeof useChatInputState>[1]
+}
+
+export function ChatInput({
+	onSendMessage,
+	waitingForResponse,
+	scrollToBottom,
+	state,
+	dispatch,
+}: ChatInputProps) {
+	const { input, images, openWhiteboard, isDragging } = state
+	const disabled = waitingForResponse || isDragging
+
+	const textareaRef = useRef<HTMLTextAreaElement>(null)
+
+	useEffect(() => {
+		if (!disabled && textareaRef.current) {
+			// focus the textarea when the input is enabled
+			textareaRef.current.focus()
+		}
+	}, [disabled])
+
+	// Auto-resize textarea and scroll to bottom when content changes.
+	useLayoutEffect(() => {
+		if (textareaRef.current) {
+			// Reset height to auto to get the correct scrollHeight
+			textareaRef.current.style.height = 'auto'
+			// Set height based on scrollHeight, with max height for ~5 lines
+			const maxHeight = 5 * 24 // Approximate line height * 5 lines
```

Comment:
to make this easier to reuse as a template, prbly you could use `5lh` if that's the intent:
https://developer.mozilla.org/en-US/docs/Web/CSS/length#lh